_posts/2019/2019-08-20-end-to-end-tests-with-geb-and-spock.md

```diff
@@ -0,0 +1,154 @@
+---
+title: End-to-end tests with Geb and Spock
+layout: post
+date: '2019-08-20'
+header_image: public/.jpg
+header_position: bottom
+header_overlay: true
+category: tech-stories
+tags: ["tech", "methods-and-tools", "coding"]
+authors: ["Thomas", "Torben"]
+about_authors: ["thirsch", "tulber"]
+---
+
+This blog post is going to be a summary of our recent endeavours to establish a solid end-to-end testing culture at ePages,
+specifically in the context of third-party service providers, e.g. payments and shippings.
+
+I will write about the motivation to start a test project, the goals that we set out to achieve, and some common problems
+and their solutions.
+
+
+## Motivation & Goals
+
+Since several years Selenium has been used at ePages. A framework was developed, with all the good intentions of categorising,
+structuring and maintaining the tests. But in the context of our third-party integrations this has been notoriously difficult.
+
+So after some discussions, deliberations, and a few Git branches later, we decided to start a new test project for this purpose,
+to be able to work on this independently, without any dependencies to other teams or existing code.
+
+We specifically did not want to create a new framework, to replace the old one.
+We wanted a way to simply and effectively develop end-to-end tests, and to be able to stabilise them.
+Last but not least we wanted to make use of Selenium Grid to parallelise execution of the tests, in order to improve the feedback
+cycle.
+
+[Geb](https://gebish.org/), [Spock](http://spockframework.org/) and [Gradle](https://gradle.org/) are a very good fit to approach
```

Comment:
Please add {:target="_blank"} behind the links

Reply:
Right, there are still some people who do not always use middle mouse button to open links...

Either way, I greatly prefer proper HTML markup in that case, instead of some markdown dialect that my editor cannot understand.